### DISCUSSION TITLE
Installing SH-RPI issue

### DISCUSSION THREAD

**Discussion by HPN-1956** [hatlabs/discussions#75] [General]
HI
In order to circumvent other issues with the SH-RPI software, I tried today to reinstall the SH-RPI software on a new image of Raspberry Pi.
The image is called Raspberry Pi OS(64 bit) released 2023-10-10.
The error occurred right after Sudo pip3 install.
"
hpn@raspberrypi:~ $ sudo apt install -y python3-pip
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
python3-pip is already the newest version (23.0.1+dfsg-1+rpt1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
hpn@raspberrypi:~ $ sudo pip3 install .
**error: externally-managed-environment**
"
The error "externally managed environment seem to be related to the new image available on Raspberry pi imager.

Please suggest a solution

**Comment by mairas** [comment]
Hi,

Thank you for the report. This is indeed an issue with Debian Bookworm. I'll fix the problem right away, but a short-term workaround is to apply the following command. That will revert the `pip` behavior back to how it was in previous Debian (or Raspberry Pi OS) versions:

    sudo rm /usr/lib/python3.11/EXTERNALLY-MANAGED

**Reply by mairas** [reply]
This change alone won't enable installation on Pi OS Bookworm due to some changes in how I2C needs to be enabled. Working on that as well.

**Comment by mairas** [comment]
I have now merged a fix for this and some other issues created by the new Pi OS version. Could you please try reinstalling and report back here?

**Reply by HPN-1956** [reply]
I have now reinstalled the sailor Hat on the latest Pi OS and the installation went smoothly without any errors. The hat works fine when power is turned on and off.
I.m still having issues with the config file though. There is no config file in the etc folder and I am not able to modify the configurable items.
I have attached the install log and some outputs from the shrpi and shrpid commands.
I hope it will show the trouble i have.

Best regards
Hans Peter
[shrpid commands.txt](https://github.com/hatlabs/discussions/files/13232649/shrpid.commands.txt)
[shrpi cmmands.txt](https://github.com/hatlabs/discussions/files/13232650/shrpi.cmmands.txt)
[SHRPI install log.txt](https://github.com/hatlabs/discussions/files/13232651/SHRPI.install.log.txt)

**Comment by HPN-1956** [comment]
SureI’l do it tomorrow  when I’m back home. Sendt fra min iPhoneMed venlig hilsenHans Peter NielsenDen 31. okt. 2023 kl. 18.00 skrev Matti Airas ***@***.***>:﻿
I have now merged a fix for this and some other issues created by the new Pi OS version. Could you please try reinstalling and report back here?
